src/overlays/actors/ovl_En_Rd/z_en_rd.c

```diff
@@ -173,83 +205,97 @@ void EnRd_Destroy(Actor* thisx, GlobalContext* globalCtx) {
     if (gSaveContext.sunsSongState != SUNSSONG_INACTIVE) {
         gSaveContext.sunsSongState = SUNSSONG_INACTIVE;
     }
+
     Collider_DestroyCylinder(globalCtx, &this->collider);
 }
 
-void func_80AE2630(GlobalContext* globalCtx, Actor* thisx, s32 arg2) {
-    Actor* enemyIt = globalCtx->actorCtx.actorLists[ACTORCAT_ENEMY].head;
-
-    while (enemyIt != NULL) {
-        if ((enemyIt->id != ACTOR_EN_RD) || (enemyIt == thisx) || (enemyIt->params < 0)) {
-            enemyIt = enemyIt->next;
+/**
+ * This function does two things depending on whether setParent is true or false.
+ * - If setParent is true, this function sets thisx to be the parent for all Redeads
+ *   in the area that are capable of mourning. This is used right when thisx first
+ *   dies to make the other Redeads mourn it.
+ * - If setParent is false, this function nulls out the parent for all Redeads in the
+ *   are whose parents is thisx. This is used when thisx is fading away to make the
+ *   other Redeads stop mourning over it.
+ */
+void EnRd_UpdateParentForOtherRedeads(GlobalContext* globalCtx, Actor* thisx, s32 setParent) {
```

Comment:
I went with `EnRd_UpdateMourningTarget`, since I feel that describes what it does precisely; it updates the target for the Redead's mourning, including the "no target" case when the third parameter is false. Let me know if you don't like the name, though.